components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/logout/util/SAMLLogoutUtil.java

```diff
@@ -0,0 +1,307 @@
+     * @param statusCode         Status Code of the Error Response.
+     * @param statusMsg          Status Message of the Error Response.
+     * @return String            Encoded Error Response.
+     * @throws SAMLIdentityException Error when building the Error Response.
+     */
+    public static String buildErrorResponse(SAMLMessageContext samlMessageContext, String
+            inResponseTo, String statusCode, String statusMsg) throws SAMLIdentityException {
+
+        try {
+            LogoutResponse errorResponse = buildResponse(samlMessageContext, inResponseTo, statusCode, statusMsg);
+            return SSOUtils.encode(SSOUtils.marshall(errorResponse));
+        } catch (SAMLSSOException e) {
+            throw new SAMLIdentityException("Error Serializing the SAML Response", e);
+        }
+    }
+
+    /**
+     * Build the Logout Response for logout request from the federated IdP.
+     *
+     * @param samlMessageContext {@link SAMLMessageContext} object which holds details on logout flow.
+     * @param inResponseTo       ID of the Logout Request.
+     * @param statusCode         Status Code of the Error Response.
+     * @param statusMsg          Status Message of the Error Response.
+     * @return Logout Response   Built Logout Response.
+     * @throws SAMLIdentityException Error when building the Logout Response.
+     */
+    public static LogoutResponse buildResponse(SAMLMessageContext samlMessageContext, String inResponseTo,
+                                               String statusCode, String statusMsg) throws SAMLIdentityException {
+
+        try {
+            doBootstrap();
+            String issuerID = (String) samlMessageContext.getFedIdPConfigs().get(SP_ENTITY_ID);
+            String acsUrl = (String) samlMessageContext.getFedIdPConfigs().get(SSO_URL);
+            String isResponseSigned = (String) samlMessageContext.getFedIdPConfigs().get(IS_AUTHN_RESP_SIGNED);
+            boolean isIncludeCert = TRUE.equals(samlMessageContext.getFedIdPConfigs().get(INCLUDE_CERT));
+
+            LogoutResponse logoutResp = new LogoutResponseBuilder().buildObject();
+            logoutResp.setID(createID());
+            logoutResp.setInResponseTo(inResponseTo);
+            logoutResp.setIssuer(getIssuer(issuerID));
+            logoutResp.setVersion(SAMLVersion.VERSION_20);
+            logoutResp.setStatus(buildStatus(statusCode, statusMsg));
+            logoutResp.setIssueInstant(new DateTime());
+            logoutResp.setDestination(acsUrl);
+
+            if (TRUE.equals(isResponseSigned) && SUCCESS_CODE.equals(statusCode)) {
+                SSOUtils.setSignature(logoutResp, null, null, isIncludeCert,
+                        new X509CredentialImpl(samlMessageContext.getTenantDomain(), null));
+            }
+            return logoutResp;
+
+        } catch (SAMLSSOException e) {
+            throw new SAMLIdentityException("Error occurred while setting the signature of logout response", e);
```

Comment:
shall we include more context info in this error message. such as the issuer?

Reply:
Fixed with PR #133